app/templates/courses/teacher-courses-view.jade

```diff
@@ -101,7 +102,7 @@ mixin course-info(course)
         i(data-i18n="courses.web_dev_language_transition")
 
     if me.isTeacher() || view.ownedClassrooms.size() || me.isAdmin()
-      - var disableButtons = !course.get('free') && !view.paidTeacher;
+      - var disableButtons = !view.paidTeacher && (features.china ? (course.get('campaignID') != "55b29efd1cd6abe8ce07db0d" && course.get('campaignID') != "5789236960deed1f00ec2ab8" && course.get('campaignID')  != "578913f2c8871ac2326fa3e4") : !course.get('free'))
```

Comment:
See my queries above.
This check is hard to understand. One option is to move this onto the User model. Then the check could be:

`- var disableButtons = me.canAccessCampaignSolutions(campaignID)`

Reply:
Then we can write tests to check this behavior is correct.

Reply:
This is a Chinese demand that teachers  and students should only allow to access that 3 campaign freely, you refer to the asana task

Reply:
Can we distinguish user's nation (Features.china)inside the user model?
If not, maybe we need pass the flag features.china to the function?